### PR TITLE
Switch to use newsletters signup on ET/Basket instead (goodbye BSD)

### PR DESCRIPTION
**Pull request by mmmavis**
- [x] change form destination
- [x] update unit tests

(mimicing https://github.com/mozilla/advocacy.mozilla.org/blob/master/components/encrypt-signup.jsx#L783)

### CODE REVIEW

File: components/newsletter-signup/SignupForm.jsx [hunk 1]
```diff
@@ -12,19 +18,43 @@ var EmailSignupForm = React.createClass({
...
+      .send({
+        newsletters: 'mozilla-learning-network',
+        source_url: encodeURIComponent(this.props.sourceUrl),
+        lang: 'en',
+        email: this.state.email,
+        trigger_welcome: 'N'
+      })
+      .end(function(err, res){
+        if ( err ) { 
+          // TODO: add error handling
```
Comment (mmmavis) [inline components/newsletter-signup/SignupForm.jsx]: suggestions? maybe we can add a followup ticket and get help from designers?

**Comment by mmmavis** [comment]
@alicoding @Pomax can i ask for a PR review plz?

File: components/newsletter-signup/SignupForm.jsx [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 var _ = require('underscore');
 
-var EmailSignupForm = React.createClass({
+var validateSignupForm = require('./validateSignupForm');
+var config = require('../../config/config');
+
+var SignupForm = React.createClass({
   mixins: [LinkedStateMixin],
+  propTypes: {
+    sourceUrl: React.PropTypes.string,
```
Comment (Pomax) [inline components/newsletter-signup/SignupForm.jsx]: `isRequired`?

File: components/newsletter-signup/SignupForm.jsx [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
+  },
   getInitialState: function() {
     return {
       email: "",
```
Comment (Pomax) [inline components/newsletter-signup/SignupForm.jsx]: should this be state, or prop? Feels like the latter.

File: components/newsletter-signup/SignupForm.jsx (hunk 1 above)
Comment (Pomax) [inline components/newsletter-signup/SignupForm.jsx]: getting design input is a good idea. What are the possible errors that can lead to this state? (i.e. how many different kinds of failures do we need to be able to present to the user)

File: components/newsletter-signup/SignupForm.jsx [hunk 4]
```diff
@@ -48,7 +78,7 @@ var EmailSignupForm = React.createClass({
           <label htmlFor={idPrefix+"privacy"} className="sr-only">
             I'm okay with you handling this info as you explain in your <a href="https://www.mozilla.org/en-US/privacy/websites/">privacy policy</a>
           </label>
-          <input id={idPrefix+"privacy"} name={process.env.NEWSLETTER_MAILINGLIST_PRIVACY_NAME} type="checkbox" className="sr-only" checked readOnly required />
```
Comment (mmmavis) [inline components/newsletter-signup/SignupForm.jsx]: ET/Basket doesn't need this.

File: components/newsletter-signup/SignupForm.jsx [hunk 5]
```diff
@@ -12,19 +18,43 @@ var EmailSignupForm = React.createClass({
...
+      .end(function(err, res){
+        if ( err ) { 
+          // TODO: add error handling
+          return; 
+        }
+
+        if ( res.statusCode === 200 ) {
+          window.location = config.ORIGIN+"?signup=thanks";
+        } else {
+          // TODO: add error handling
```
Comment (Pomax) [inline components/newsletter-signup/SignupForm.jsx]: this should be part of the above-mentioned followup

File: components/newsletter-signup/SignupForm.jsx (hunk 2 above)
Comment (mmmavis) [inline components/newsletter-signup/SignupForm.jsx]: oops, nice catch!

File: components/newsletter-signup/SignupForm.jsx (hunk 3 above)
Comment (mmmavis) [inline components/newsletter-signup/SignupForm.jsx]: `LinkedStateMixin` is used here. So I think `state` is more suitable here? :grin: (see [two-way binding helper](https://facebook.github.io/react/docs/two-way-binding-helpers.html))

**Pull request merged**